pkg/services/login/authinfo.go

```diff
@@ -114,6 +114,19 @@ func IsExternallySynced(cfg *setting.Cfg, authModule string) bool {
 	return true
 }
 
+func IsGrafanaAdminExternallySynced(cfg *setting.Cfg, authModule string, oAuthAndAllowAssignGrafanaAdmin bool) bool {
+	if !IsExternallySynced(cfg, authModule) {
+		return false
+	}
+
+	switch authModule {
+	case JWTModule:
+		return cfg.JWTAuthAllowAssignGrafanaAdmin
+	default:
+		return oAuthAndAllowAssignGrafanaAdmin
+	}
```

Comment:
question: am I correct in assuming that Grafana Admin role can only be synced for JWT and OAuth providers?

I'm also not keen on the variable naming here, but couldn't come up with anything better.

Reply:
I think only JWT, OAuth and LDAP

Reply:
Thanks for the heads up about LDAP, I had missed that one.